Assets/Colyseus/Runtime/Scripts/Models/ColyseusClient.cs

```diff
@@ -28,22 +28,81 @@ public class ColyseusClient
+		/// The <see cref="ColyseusSettings"/> currently assigned to this client object
+		/// </summary>
+		private ColyseusSettings _colyseusSettings;
+
+		/// <summary>
+		///     Occurs when the <see cref="ColyseusClient" /> successfully connects to the <see cref="ColyseusRoom{T}" />.
+		/// </summary>
+		public static event ColyseusAddRoomEventHandler onAddRoom;
+
+		/// <summary>
+		/// The getter for the <see cref="ColyseusSettings"/> currently assigned to this client object
+		/// </summary>
+		public ColyseusSettings Settings
+        {
+	        get
+	        {
+		        return _colyseusSettings;
+	        }
+
+	        private set
+	        {
+		        _colyseusSettings = value;
+
+		        // Instantiate our ColyseusRequest object with the settings object
+				colyseusRequest = new ColyseusRequest(_colyseusSettings);
+	        }
+        }
+
+        /// <summary>
+		/// Object to perform <see cref="UnityEngine.Networking.UnityWebRequest"/>s to the server.
+		/// </summary>
+        public ColyseusRequest colyseusRequest;
+
         /// <summary>
         ///     Initializes a new instance of the <see cref="ColyseusClient" /> class with
-        ///     the specified Colyseus Game Server Server endpoint.
+        ///     the specified Colyseus Game Server endpoint.
         /// </summary>
         /// <param name="endpoint">
         ///     A <see cref="string" /> that represents the WebSocket URL to connect.
         /// </param>
         public ColyseusClient(string endpoint)
         {
             Endpoint = new UriBuilder(new Uri(endpoint));
+
+            // Create ColyseusSettings object to pass to the ColyseusRequest object
+            ColyseusSettings settings = new ColyseusSettings()
+	            {
+		            colyseusServerAddress = Endpoint.Host,
+		            colyseusServerPort = Endpoint.Port.ToString(),
+					useSecureProtocol = Endpoint.ToString().StartsWith("wss") || Endpoint.ToString().StartsWith("https")
+	            };
+
+            Settings = settings;
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ColyseusClient"/> class with
+		/// the specified Colyseus Settings object.
+		/// </summary>
+		/// <param name="settings">The settings you wish to use</param>
+		/// <param name="useWebSocketEndpoint">Determines whether the connection endpoint should use either web socket or http protocols.</param>
+		public ColyseusClient(ColyseusSettings settings, bool useWebSocketEndpoint)
+        {
+	        Endpoint = new UriBuilder(new Uri(useWebSocketEndpoint ? settings.WebSocketEndpoint : settings.WebRequestEndpoint));
+
+	        Settings = settings;
         }
 
-        /// <summary>
-        ///     Occurs when the <see cref="ColyseusClient" /> successfully connects to the <see cref="ColyseusRoom{T}" />.
-        /// </summary>
-        public static event ColyseusAddRoomEventHandler onAddRoom;
+		public void OverrideSettings(ColyseusSettings newSettings, bool useWebSocketEndpoint)
```

Comment:
I'd suggest renaming `OverrideSettings` to `SetSettings`, and call it also during `ColyseusClient`'s constructor

Reply:
hmm yeah can do! I can see why it would be attractive to call it within one of the ColyseusClient's constructors. I wouldn't want to do it in both constructors though. The only reservation I have about doing so is the constructor that only takes a string endpoint depends on the UriBuilder Endpoint to correctly parse out the components of the endpoint in order to construct the ColyseusSettings object at runtime.  We wouldn't want to then call SetSettings where Endpoint would get recreated using the parsed data from the settings object.  
I suppose we could, it just seems redundant and then we might lose data from the original endpoint in UriBuilder.